server/src/main/java/org/elasticsearch/index/mapper/LegacyGeoShapeFieldMapper.java

```diff
@@ -452,6 +452,10 @@ public LegacyGeoShapeFieldMapper(String simpleName, MappedFieldType mappedFieldT
         super(simpleName, mappedFieldType, Collections.singletonMap(mappedFieldType.name(), Lucene.KEYWORD_ANALYZER),
             builder.ignoreMalformed.get(), builder.coerce.get(), builder.ignoreZValue.get(), builder.orientation.get(),
             multiFields, copyTo, indexer, parser);
+        if (builder.indexCreatedVersion.onOrAfter(Version.V_8_0_0)) {
+            throw new IllegalArgumentException("mapper [" + name()
+                + "] of type [geo_shape] with deprecated parameters is no longer allowed");
```

Comment:
It would be good to be a bit clearer about what is causing the error here.  Maybe instead of having the version check here we should move it to `containsDeprecatedParameter` so that the error message can explicitly contain the parameters that were used and are no longer allowed?

Reply:
What I have done is to collect the used deprecated parameters in the builder so we can report back to the user. wdyt?